spec/handlers/sessions_callback_spec.rb

```diff
@@ -4,237 +4,308 @@
   
   let(:user_state) { MockUserState.new }
 
-  context "when not signed in and no existing auth" do
```

Comment:
Yes. I tried to nest them somewhat so we know which specs are related.

This particular spec ended up becoming the very first spec in the file after the change.

All the code got copied, however... and I found a feel untested cases that I marked as pending with `xit`.